### PR TITLE
Delete some `if __name__ == "__main__"` clauses.

### PR DESCRIPTION
**Pull request by anntzer**
In offsetbox and mpl_axes, they are just examples (mpl_axes doesn't even
call `show()`); if anything the examples belong in the examples/
directory but they don't look particularly worth moving there either.

In rcsetup, the clause is checking that the defaults in rcParams don't
get modified by the validators.  If anything this should be a test, but
it doesn't look like it mattered.  I did fix `text.latex.preamble`,
`pgf.preamble`, and `keymap.fullscreen` to be idempotent under the
validator; now the only rcParam that gets changed is `savefig.bbox` (due
to the presence of the "standard" / None synonym) but again this isn't a
problem.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/rcsetup.py [hunk 1]
```diff
@@ -1115,7 +1115,7 @@ def _validate_linestyle(ls):
     'text.color':          ['black', validate_color],
     'text.usetex':         [False, validate_bool],
     'text.latex.unicode':  [True, validate_bool],
-    'text.latex.preamble': [[''], validate_stringlist],
+    'text.latex.preamble': [[], validate_stringlist],
```
Comment (QuLogic) [inline lib/matplotlib/rcsetup.py]: This change appears unrelated.

Comment (anntzer) [reply lib/matplotlib/rcsetup.py]: That's because `matplotlib.rcsetup.validate_stringlist([""]) == []` so the code I deleted complained that the validator was not idempotent here.
